sdks/go/pkg/beam/runners/dataflow/dataflowlib/job.go

```diff
@@ -140,6 +144,18 @@ func Translate(p *pb.Pipeline, opts *JobOptions, workerURL, jarURL, modelURL str
 	if opts.NumWorkers > 0 {
 		job.Environment.WorkerPools[0].NumWorkers = opts.NumWorkers
 	}
+	if opts.Algorithm != "" {
+		settings := &df.AutoscalingSettings{
+			Algorithm: map[string]string{
+				"NONE":             "AUTOSCALING_ALGORITHM_NONE",
+				"THROUGHPUT_BASED": "AUTOSCALING_ALGORITHM_BASIC",
+			}[opts.Algorithm],
+		}
+		if opts.MaxNumWorkers > 0 {
```

Comment:
`MaxNumWorkers` could be set even if `opts.Algorithm` is not set. If Algorithm is not set, Dataflow will choose the appropriate autoscaling algortihm and max num workers would still be used for that.